### PR TITLE
add namespace declaration to parser

### LINKED ISSUE #128: specify a namespace for an RSDL file
I like it to be able to specify a namespace. 

It has a bit of a larger impact since I believe we a) need to be able to fully qualify (type-) names and b) import/reference other namespaces. 

I would suggest to not specify the namespace as an annotation and not so deep into the document.
I would suggest 
a) one namespace per document and 
b) an (optional ) declaration at the top of the document namespace "hello"  that specifies the namespace name. No needs for any braces since this is for the complete document. I believe that simplifies the syntax and makes it less obtrustive.

Happy to work on the namespace declaration and leave the reference/fqn design for later.

_Originally posted by @chrisspre in https://github.com/oasis-open/odata-rapid/issues/121#issuecomment-702375681_

### PR DESCRIPTION
**Pull request by chrisspre**
- add namespace declaration type to AST
- extend parser to read namespace declaration before any of the other schema elements
- fix transformation to CSDL to leverage declared namespace name if present.

fixes #128

### CODE REVIEW

File: tools/rsdl/dotnet/rapid/sample.rsdl [hunk 1]
```diff
@@ -1,3 +1,5 @@
+namespace "jetsons"
```
Comment (ralfhandl) [inline tools/rsdl/dotnet/rapid/sample.rsdl:1]: I'd prefer no quotes, namespaces follow the identifier rules, they are not string values

Comment (ralfhandl) [inline tools/rsdl/dotnet/rapid/sample.rsdl:1]:
```suggestion
namespace jetsons
```

**Review by ralfhandl — CHANGES_REQUESTED** [review]

**Comment by wtrocki** [comment]
I tend to force that new change to the code requires some minimum documentation. It can have tons of TODO but the only occurrence of namespaces we will get after the merge will be sample rsdl. Adding placeholder readme is advised. 

Merged suggestion to see how this will work.
Compiled code and played with it locally.

**Comment by wtrocki** [comment]
Obviously, changes will be required after the merge to drop `"`. from namespace.

**Comment by ralfhandl** [comment]
#132 extends ABNF with namespace (and no quotes)

File: tools/rsdl/dotnet/rapid/sample.rsdl (hunk 1 above)
Comment (chrisspre) [reply tools/rsdl/dotnet/rapid/sample.rsdl]: Understood. Will fix today.

**Comment by chrisspre** [comment]
> I tend to force that new change to the code requires some minimum documentation. It can have tons of TODO but the only occurrence of namespaces we will get after the merge will be sample rsdl. Adding placeholder readme is advised.

Makes sense. I will make an effort to make that part of the changes going forward. At the moment the documentation is a bit fluid and I don't want to step over Mike's changes. That is not to object, just to say it might take a little to get into the rhythm.

**Comment by chrisspre** [comment]
Made the change to `namespace foo.bar.baz` in the latest commit. 
Not sure what I need to do in the github UI to resolve the "Changes requested" status

**Pull request merged**